moving a local object in a return statement prevents copy elision

At least with `-Wall -Wextra` you'll get a lot of warnings like this:

```
/Users/aklimov/NET/WS/icinga2/lib/perfdata/influxdbwriter.cpp:37:9: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
        return std::move(request);
               ^
/Users/aklimov/NET/WS/icinga2/lib/perfdata/influxdbwriter.cpp:37:9: note: remove std::move call here
```

It already tells the solution: remove std::move call here

--- a/lib/base/netstring.cpp
+++ b/lib/base/netstring.cpp
@@ -194,7 +194,7 @@ String NetString::ReadStringFromStream(const Shared<AsioTlsStream>::Ptr& stream,
 		BOOST_THROW_EXCEPTION(std::invalid_argument("Invalid NetString (missing ,)"));
 	}
 
-	return std::move(payload);
+	return payload;
 }
 
 /**
@@ -273,7 +273,7 @@ String NetString::ReadStringFromStream(const Shared<AsioTlsStream>::Ptr& stream,
 		BOOST_THROW_EXCEPTION(std::invalid_argument("Invalid NetString (missing ,)"));
 	}
 
-	return std::move(payload);
+	return payload;
 }
 
 /**

--- a/lib/base/perfdatavalue.cpp
+++ b/lib/base/perfdatavalue.cpp
@@ -154,7 +154,7 @@ static const UoMs l_CsUoMs (([]() -> UoMs {
 		}
 	}
 
-	return std::move(uniqUoms);
+	return uniqUoms;
 })());
 
 static const UoMs l_CiUoMs (([]() -> UoMs {
@@ -208,7 +208,7 @@ static const UoMs l_CiUoMs (([]() -> UoMs {
 		}
 	}
 
-	return std::move(uniqUoms);
+	return uniqUoms;
 })());
 
 PerfdataValue::PerfdataValue(const String& label, double value, bool counter,

--- a/lib/base/tlsutility.cpp
+++ b/lib/base/tlsutility.cpp
@@ -298,7 +298,7 @@ Shared<boost::asio::ssl::context>::Ptr SetupSslContext(String certPath, String k
 		}
 	}
 
-	return std::move(context);
+	return context;
 }
 
 /**

--- a/lib/cli/consolecommand.cpp
+++ b/lib/cli/consolecommand.cpp
@@ -557,7 +557,7 @@ Shared<AsioTlsStream>::Ptr ConsoleCommand::Connect()
 		throw;
 	}
 
-	return std::move(stream);
+	return stream;
 }
 
 /**

--- a/lib/icinga/checkable-comment.cpp
+++ b/lib/icinga/checkable-comment.cpp
@@ -54,7 +54,7 @@ Comment::Ptr Checkable::GetLastComment() const
 		}
 	}
 
-	return std::move(lastComment);
+	return lastComment;
 }
 
 void Checkable::RegisterComment(const Comment::Ptr& comment)

--- a/lib/icingadb/icingadb-objects.cpp
+++ b/lib/icingadb/icingadb-objects.cpp
@@ -554,7 +554,7 @@ std::vector<std::vector<intrusive_ptr<ConfigObject>>> IcingaDB::ChunkObjects(std
 		chunks.emplace_back(offset, end);
 	}
 
-	return std::move(chunks);
+	return chunks;
 }
 
 void IcingaDB::DeleteKeys(const RedisConnection::Ptr& conn, const std::vector<String>& keys, RedisConnection::QueryPriority priority) {
@@ -591,7 +591,7 @@ std::vector<String> IcingaDB::GetTypeOverwriteKeys(const String& type)
 		keys.emplace_back(m_PrefixConfigCheckSum + type + ":argument");
 	}
 
-	return std::move(keys);
+	return keys;
 }
 
 std::vector<String> IcingaDB::GetTypeDumpSignalKeys(const Type::Ptr& type)
@@ -621,7 +621,7 @@ std::vector<String> IcingaDB::GetTypeDumpSignalKeys(const Type::Ptr& type)
 		keys.emplace_back(m_PrefixConfigObject + lcType + ":argument");
 	}
 
-	return std::move(keys);
+	return keys;
 }
 
 template<typename ConfigType>

--- a/lib/icingadb/redisconnection.hpp
+++ b/lib/icingadb/redisconnection.hpp
@@ -598,7 +598,7 @@ std::vector<char> RedisConnection::ReadLine(AsyncReadStream& stream, boost::asio
 
 		if (next == '\r') {
 			asio::async_read(stream, buf, yc);
-			return std::move(line);
+			return line;
 		}
 
 		line.emplace_back(next);

--- a/lib/perfdata/elasticsearchwriter.cpp
+++ b/lib/perfdata/elasticsearchwriter.cpp
@@ -646,7 +646,7 @@ OptionalTlsStream ElasticsearchWriter::Connect()
 		}
 	}
 
-	return std::move(stream);
+	return stream;
 }
 
 void ElasticsearchWriter::AssertOnWorkQueue()

--- a/lib/perfdata/influxdb2writer.cpp
+++ b/lib/perfdata/influxdb2writer.cpp
@@ -27,7 +27,7 @@ boost::beast::http::request<boost::beast::http::string_body> Influxdb2Writer::As
 
 	request.set(boost::beast::http::field::authorization, "Token " + GetAuthToken());
 
-	return std::move(request);
+	return request;
 }
 
 Url::Ptr Influxdb2Writer::AssembleUrl()
@@ -40,5 +40,5 @@ Url::Ptr Influxdb2Writer::AssembleUrl()
 	url->AddQueryElement("org", GetOrganization());
 	url->AddQueryElement("bucket", GetBucket());
 
-	return std::move(url);
+	return url;
 }

--- a/lib/perfdata/influxdbcommonwriter.cpp
+++ b/lib/perfdata/influxdbcommonwriter.cpp
@@ -192,7 +192,7 @@ OptionalTlsStream InfluxdbCommonWriter::Connect()
 		}
 	}
 
-	return std::move(stream);
+	return stream;
 }
 
 void InfluxdbCommonWriter::CheckResultHandler(const Checkable::Ptr& checkable, const CheckResult::Ptr& cr)
@@ -540,7 +540,7 @@ boost::beast::http::request<boost::beast::http::string_body> InfluxdbCommonWrite
 	request.body() = std::move(body);
 	request.content_length(request.body().size());
 
-	return std::move(request);
+	return request;
 }
 
 Url::Ptr InfluxdbCommonWriter::AssembleBaseUrl()
@@ -552,7 +552,7 @@ Url::Ptr InfluxdbCommonWriter::AssembleBaseUrl()
 	url->SetPort(GetPort());
 	url->AddQueryElement("precision", "s");
 
-	return std::move(url);
+	return url;
 }
 
 void InfluxdbCommonWriter::ValidateHostTemplate(const Lazy<Dictionary::Ptr>& lvalue, const ValidationUtils& utils)

--- a/lib/perfdata/influxdbwriter.cpp
+++ b/lib/perfdata/influxdbwriter.cpp
@@ -34,7 +34,7 @@ boost::beast::http::request<boost::beast::http::string_body> InfluxdbWriter::Ass
 		);
 	}
 
-	return std::move(request);
+	return request;
 }
 
 Url::Ptr InfluxdbWriter::AssembleUrl()
@@ -52,5 +52,5 @@ Url::Ptr InfluxdbWriter::AssembleUrl()
 	if (!GetPassword().IsEmpty())
 		url->AddQueryElement("p", GetPassword());
 
-	return std::move(url);
+	return url;
 }

--- a/lib/remote/configobjectutility.cpp
+++ b/lib/remote/configobjectutility.cpp
@@ -53,7 +53,7 @@ String ConfigObjectUtility::GetObjectConfigPath(const Type::Ptr& type, const Str
 	 * creation early so that configuration inconsistencies throughout the cluster are avoided.
 	 */
 	if ((type->GetName() != "Comment" && type->GetName() != "Downtime") || Utility::PathExists(longPath)) {
-		return std::move(longPath);
+		return longPath;
 	}
 
 	/* Maximum length 80 bytes object name + 3 bytes "..." + 40 bytes SHA1 (hex-encoded) */

--- a/lib/remote/eventqueue.cpp
+++ b/lib/remote/eventqueue.cpp
@@ -231,7 +231,7 @@ Dictionary::Ptr EventsInbox::Shift(boost::asio::yield_context yc, double timeout
 
 	auto event (std::move(m_Queue.front()));
 	m_Queue.pop();
-	return std::move(event);
+	return event;
 }
 
 EventsSubscriber::EventsSubscriber(std::set<EventType> types, String filter, const String& filterSource)

--- a/lib/remote/jsonrpc.cpp
+++ b/lib/remote/jsonrpc.cpp
@@ -113,7 +113,7 @@ String JsonRpc::ReadMessage(const Shared<AsioTlsStream>::Ptr& stream, ssize_t ma
 		std::cerr << ConsoleColorTag(Console_ForegroundBlue) << "<< " << jsonString << ConsoleColorTag(Console_Normal) << "\n";
 #endif /* I2_DEBUG */
 
-	return std::move(jsonString);
+	return jsonString;
 }
 
 /**
@@ -134,7 +134,7 @@ String JsonRpc::ReadMessage(const Shared<AsioTlsStream>::Ptr& stream, boost::asi
 		std::cerr << ConsoleColorTag(Console_ForegroundBlue) << "<< " << jsonString << ConsoleColorTag(Console_Normal) << "\n";
 #endif /* I2_DEBUG */
 
-	return std::move(jsonString);
+	return jsonString;
 }
 
 /**

--- a/plugins/check_nscp_api.cpp
+++ b/plugins/check_nscp_api.cpp
@@ -206,7 +206,7 @@ static Shared<AsioTlsStream>::Ptr Connect(const String& host, const String& port
 		throw;
 	}
 
-	return std::move(stream);
+	return stream;
 }
 
 static const char l_ReasonToInject[2] = {' ', 'X'};
@@ -223,7 +223,7 @@ boost::asio::mutable_buffer GetFirstNonZeroBuffer(const MutableBufferSequence& m
 		asio::mutable_buffer buf (*current);
 
 		if (buf.size() > 0u) {
-			return std::move(buf);
+			return buf;
 		}
 	}
 

